domain/secret/state/state.go

```diff
@@ -1510,34 +1512,39 @@ WHERE  rev.secret_id = $secretRevision.secret_id AND rev.revision = $secretRevis
 	}, nil
 }
 
-// checkExistsIfLocal returns an error satisfying [secreterrors.SecretNotFound] if the specified
+// checkExistsIfLocal returns true of the secret is local to this model
+// It returns an error satisfying [secreterrors.SecretNotFound] if the specified
 // secret URI is from this model and the secret it refers to does not exist in the model.
-func (st State) checkExistsIfLocal(ctx context.Context, tx *sqlair.TX, uri *coresecrets.URI) error {
+func (st State) checkExistsIfLocal(ctx context.Context, tx *sqlair.TX, uri *coresecrets.URI) (bool, error) {
 	query := `
-SELECT ok as &M.ok FROM (
-    SELECT True as ok FROM secret_metadata sm
+WITH local AS
+    (SELECT 'local' AS is_local FROM secret_metadata sm
     WHERE 
         (EXISTS (SELECT uuid FROM model WHERE uuid = $secretRef.source_uuid) OR $secretRef.source_uuid = '')
-        AND sm.secret_id = $secretRef.secret_id
-    UNION
-    SELECT True as ok FROM model
-    WHERE
-        NOT EXISTS (SELECT uuid FROM model WHERE uuid = $secretRef.source_uuid)
-)
+        AND sm.secret_id = $secretRef.secret_id),
```

Comment:
We can use CASE, THEN.
Also, I don't think we need the is_remote check at all.

Reply:
The method needs to be able to determine if the secret exists in the secret metadata table AND if the source model in the URI matches the current model (if not empty). I'm not sure how we'd be able to determine both pieces of information with just the one query.
I've pushed a change to make the SQL better.